### ISSUE TITLE
Trigger onError hook while sending html error page to user

### ISSUE THREAD

**Issue by nckcol** [fastify/fastify#2346]
#### You have already researched for similiar issues?
yes

#### What are you trying to achieve or the steps to reproduce?
I want to log errors while sending html error to the user with error statusCode (>= 500).
Purpose of `onError` hook, described in docs, totally covers my case:
> This hook is useful if you need to do some custom error logging ...

But there is problem with "sends an error back to the user":
> This hook will be executed only after the customErrorHandler has been executed, and only if the customErrorHandler sends an error back to the user ...

So the question is how to create such response in `errorHandler` that is an Error (triggers onError hook for logging purposes) and responds with user friendly error page (but still with error status code) at the same time.
I can do such logging with `onResponse` hook for sure. But it will be better if i could use onError hook

```js
// global for all subapps error logging
fastify.addHook('onError', (request, reply, error, done) => {
  if(error.statusCode >= 500) {
    // i want to log all application errors here
    log.error(error.statusCode, error.message)
  }
  done();
})

// user accessible routes (html sub-app)
fastify.register((instance, options, done) => {
    // error handler for user-friendly error messages
    instance.setErrorHandler((error, request, reply, done) => {
      reply.code(500).type('text/html').send(fancyHtmlError);

      // also tried this:
      // error.message = fancyHtmlError;
      // reply.code(500).type('text/html').send(error)
      // but it serializes as JSON

      // it would be nice to have something like this:
      // reply.error(error).type('text/html').send(fancyHtmlError);
      done();
    });
    instance.get('/some-route', async () => {
      const error = new Error();
      error.message = "I want to log this message"
      error.statusCode = 500;
      throw error;
    });
    done();
});

// other sub-apps with their own errorHandlers
```

#### What was the result you received?
I can't use onError hook for logging if i want to respond with user-friendly errors

#### What did you expect?
I expect a way to define that response is an error for logging with `onError` hook, but still reply with html for users, instead of json.

#### Context

* *node version*: 12.13.0
* *fastify version*: 2.14.1
* *os*: Ubuntu 20.04 LTS 64-bit

Thank you!

**Comment by jstayton** [comment]
This has also been quite confusing to me.

I've registered a custom error handler with `setErrorHandler` because I want to reply with JSON that's different than the default error handler. I expected the `onError` hook to still be called, but it seems to only be called when you `reply.send` with an actual `Error` (as the default error handler does) – right?

In other words, the `onError` hook _isn't_ triggered if you want to reply with anything other than the default serialization of an `Error`.
